### PR TITLE
Sharing a request with yourself shows a misleading error message

### PR DESCRIPTION
**Pull request by sebbi08**
# Readiness checklist

-   [ ] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

The error used to be: "There is no relationship to x".

### CODE REVIEW

File: packages/consumption/src/consumption/ConsumptionCoreErrors.ts [hunk 1]
```diff
@@ -310,6 +310,10 @@ class Requests {
         return new ApplicationError("error.consumption.requests.validation.inheritedFromItem", message);
     }
 
+    public cannotShareARequestWithYourself() {
```
Comment (Milena-Czierlinski) [inline packages/consumption/src/consumption/ConsumptionCoreErrors.ts:313]: I think usually we don't have an "a" in the name of errors e.g. we have cannotShareRelationshipAttributeOfPendingRelationship above and not cannotShare**A**RelationshipAttributeOf**A**PendingRelationship. So, I'd rather name it "cannotShareRequestWithYourself". But that's just a detail, I'm not sure how strict we are with this in general.

**Comment by Milena-Czierlinski** [comment]
Also, if you want to be thorough you could add a test for this. ;)

File: packages/consumption/src/modules/requests/outgoing/OutgoingRequestsController.ts
Comment (jkoenig134) [inline packages/consumption/src/modules/requests/outgoing/OutgoingRequestsController.ts:1]: I am missing newlines here..

**Comment by jkoenig134** [comment]
I am missing a test here.

**Review by jkoenig134 — DISMISSED** [review]

File: packages/consumption/src/consumption/ConsumptionCoreErrors.ts [hunk 2]
```diff
@@ -310,6 +310,10 @@ class Requests {
         return new ApplicationError("error.consumption.requests.validation.inheritedFromItem", message);
     }
 
+    public cannotShareRequestWithYourself() {
+        return new CoreError("error.consumption.requests.cannotShareARequestWithYourself", "You cannot share a Request with yourself.");
```
Comment (Milena-Czierlinski) [inline packages/consumption/src/consumption/ConsumptionCoreErrors.ts:314]: You need to also remove the "A" from the error code.

**Review by Milena-Czierlinski — CHANGES_REQUESTED** [review]

**Review by Milena-Czierlinski — APPROVED** [review]

**Pull request merged**